### ISSUE TITLE
psarips.com - custom rule

### ISSUE THREAD

**Issue by Bohdan-SUP** [AdguardTeam/AdguardFilters#3182]
Using adguard extension 2.4.14 with latest chrome stable

Here are extensions' settings page: https://i.imgur.com/vaEPwjZ.png

As you can see, there are no filter subscriptions, only one custom filter rule: 

**/*$domain=psarips.com,script,third-party**

Here is the filtering log of the page with the above filter rule: https://i.imgur.com/msfGHNn.png
on the following page:
https://psarips.com/movie/blood-father-2016/

It doesn't block youtube JS as a third party request which falls within the criteria of my custom rule. Then why isn't it blocking it?

User ID **1023133**, ticket SFA-382-22996

**Comment by Alex-302** [comment]
Because referer is different. User should to block youtube.com
```
||ytimg.com^$domain=youtube.com,script
```

**Comment by The-Commissioner** [comment]
I think the rule would be
||ytimg.com^$domain=psarips.com,script instead of ||ytimg.com^$domain=youtube.com,script which will otherwise completely break youtube everywhere which was not intended. ;)

**Comment by Alex-302** [comment]
Embed Youtube player is loaded as iframe. All requests in this iframe has referer=youtube.com
Look at this:
<details>
![image](http://imagizer.imageshack.com/img923/6950/rp1I4L.png)
</details>
So we can block ytimg.com only for youtube.com.

**Comment by The-Commissioner** [comment]
@Alex-302 I totally understood the first time you told me it has different referer. I was just enquiring about the rule. And remember my settings in first post of this issue. With your rule youtube.com site just displays a blank black frame and doesn't show/play videos.

**Comment by Alex-302** [comment]
Yes, my rule will break youtube.
Youtube is not loaded as JS, so your rule doesn't block it.
What is expected behawiour for you?

**Comment by The-Commissioner** [comment]
>  Youtube is not loaded as JS, so your rule doesn't block it.

Yes I gathered that too while I was testing your rule, I tested my rule also and it also failed. :P
Actually I can block YT on that site by using the simple rule ||youtube.com^$domain=psarips.com
But what **I initially wanted to do was block all third-party js** on that page which is not happening.
Like you told if there is a different referer like in this case(youtube.com), so now I'm wondering how to block these cases.
